week3/predict_citibike.Rmd

```diff
@@ -0,0 +1,176 @@
+  geom_line() +
+  scale_x_continuous(breaks=K) +
+  xlab('Polynomial Degree') +
+  ylab('RMSE')
+
+
+```
+
+Yeah these are pretty similar results, though I'm not sure why the lables are reversed. Now I'm gonna do my thing with the linear model.
+
+So now let's include all of the other columns
+
+```{r}
+  
+  #since tmin and tmax are probably super correlated, i'll only include tmax
+  # fit on the training data
+model <- lm(num_trips ~ tmax + snwd + prcp + snow + is_holiday + weekday, data=trips_per_day_train)
+    
+  # rmse on the training data
+train_err <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+
+  # rmse on the validate data
+validate_err <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+
+train_err
+validate_err
+
+```
+Great. We seem to have a lower validation error than the lowest point on the previous graph.
+
+```{r}
+tidy(model)
+```
+so according to this tidy model. The p-values are pretty decent but for a few. With an alpha of .05, snow, and every weekday don't pass the null hypothesis. Let's at least drop snow.
+
+```{r}
+model <- lm(num_trips ~ tmax + snwd + prcp + is_holiday + weekday, data=trips_per_day_train)
+    
+  # rmse on the training data
+train_err <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+
+  # rmse on the validate data
+validate_err <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+
+train_err
+validate_err
+```
+okay so it seems the error went up without snow. Thisd being said, it's probably best to keep it.
+
+also it's likely that temperature has more of a parabolic effect, since if it's really high or low, people won't ride as much, where closer to center will have the most people riding
+```{r}
+
+model <- lm(num_trips ~ poly(tmax, 2, raw = T)
+ + snwd + snow + prcp + is_holiday + weekday, data=trips_per_day_train)
+    
+  # rmse on the training data
+train_err <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+
+  # rmse on the validate data
+validate_err <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+
+train_err
+validate_err
+```
+okay, it's slightly higher. So we'll keep it like this and try it with the test data
+```{r}
+trips_per_day_train <- trips_per_day_train %>%
+  add_predictions(model) %>%
+  mutate(split = "train")
+trips_per_day_validate <- trips_per_day_validate %>%
+  add_predictions(model) %>%
+  mutate(split = "validate")
+plot_data <- bind_rows(trips_per_day_train, trips_per_day_validate)
+
+ggplot(plot_data, aes(x = ymd, y = num_trips)) +
+  geom_point(aes(color = split)) +
+  geom_line(aes(y = pred)) +
+  xlab('Day') +
+  ylab('Number of Trips') 
+```
+
+This looks really overfit. But I think this is due to the inclusion of weekdays factor. Since travel would go up and down based on if it's a weekday or day.

```

Comment:
It probably only looks overfit on this kind of plot because we're plotting with respect to time, and time isn't part of the model (really, the `weekday` variable is time-based, but as you observed, it just makes the plot look more sporadic.)